### PR TITLE
Added Phalcon\Validation\Validator\Callback

### PR DESCRIPTION
**Pull request by Jurigag**
- [x] Make sure that the PR passes in Travis CI to make the process more efficient.
- [x] Only use tabs for indentation.
- [x] if needed, rebase to the proper branch before submitting your pull request.
    If it doesn't merge cleanly with master you may be asked to rebase your changes.
- [x] Don't put submodule updates in your pull request unless they are to landed commits.
- [x] Add tests relevant to the fixed bug or new feature. See our testing guide for
    more information.
- [x] Phalcon 2 is written in Zephir, please do not submit commits that modify C generated
    files directly or those whose functionality/fixes are implemented in the C
    programming language
- [x] Remove any change to ext/kernel / *.zep.c / *.zep.h files before submitting the PR

### CODE REVIEW

**Comment by Jurigag** [comment]
Well im hardly thinking about logic of this - should message from `Callback` validator should be added if there is true returned or false ?

File: phalcon/validation/validator/callback.zep [hunk 1]
```diff
@@ -0,0 +1,126 @@
+
+/*
+ +------------------------------------------------------------------------+
+ | Phalcon Framework                                                      |
+ +------------------------------------------------------------------------+
+ | Copyright (c) 2011-2016 Phalcon Team (https://phalconphp.com)       |
```
Comment (sergeyklay) [inline phalcon/validation/validator/callback.zep]: Could you please fix indents?

File: phalcon/validation/validator/callback.zep [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+ * use Phalcon\Validation\Validator\Callback as CallbackValidator;
+ * use Phalcon\Validation\Validator\Numericality as NumericalityValidator;
+ *
+ * $validator->add(
+ *     ["user", "admin"],
+ *     new CallbackValidator(
+ *         [
+ *             "message" => "There must be only an user or admin set",
+ *             "callback" => function($data) {
+ *                 if(!empty($data->getUser()) && !empty($data->getAdmin())) {
```
Comment (sergeyklay) [inline phalcon/validation/validator/callback.zep]: `if(` => `if (`

File: phalcon/validation/validator/callback.zep [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+ *                     return false
```
Comment (sergeyklay) [inline phalcon/validation/validator/callback.zep]: `false;`

File: phalcon/validation/validator/callback.zep [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+ *         ]
+ *     )
+ * );
+ *
+ * $validator->add(
+ *     "amount",
+ *     new CallbackValidator(
+ *         [
+ *             "callback" => function($data) {
+ *                 if(!empty($data->getProduct())) {
```
Comment (sergeyklay) [inline phalcon/validation/validator/callback.zep]: `if (`

File: phalcon/validation/validator/callback.zep [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                	validation->appendMessage(new Message(strtr(message, replacePairs), field, "Callback", code));
+                	return false;
+                }
+
+                return true;
+			}
+			elseif typeof returnedValue == "object" && returnedValue instanceof Validator {
+				return returnedValue->validate(validation, field);
+			}
+			else {
```
Comment (sergeyklay) [inline phalcon/validation/validator/callback.zep]: else not needed

File: phalcon/validation/validator/exception.zep [hunk 6]
```diff
@@ -0,0 +1,31 @@
+
+/*
+ +------------------------------------------------------------------------+
+ | Phalcon Framework                                                      |
+ +------------------------------------------------------------------------+
+ | Copyright (c) 2011-2016 Phalcon Team (https://phalconphp.com)       |
```
Comment (sergeyklay) [inline phalcon/validation/validator/exception.zep]: indents

File: phalcon/validation/validator/exception.zep [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+ +------------------------------------------------------------------------+
+ */
+
+namespace Phalcon\Validation\Validator;
+
+/**
+ * Phalcon\Validation\Exception
+ *
+ * Exceptions thrown in Phalcon\Validation\Validator\* classes will use this class
+ *
```
Comment (sergeyklay) [inline phalcon/validation/validator/exception.zep]: extra line

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Comment by Jurigag** [comment]
@sergeyklay ready to merge ? or anything more to do here ?

**Comment by sergeyklay** [comment]
Thanks

**Comment by sergeyklay** [comment]
@Jurigag Could you please submit a PR to the docs repo?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,6 @@
+# [3.1.0](https://github.com/phalcon/cphalcon/releases/tag/v3.1.0) (2016-XX-XX)``
```
Comment (Jurigag) [inline CHANGELOG.md]: Oh i added not needed 
```
``
```
here @sergeyklay

Comment (sergeyklay) [inline CHANGELOG.md]: WIll be fixed in nex PR

**Pull request merged**